diff --git a/lib/node/chunk-CVV3L375.mjs b/lib/node/chunk-CVV3L375.mjs
index c37d61d709235efebe4ea19703a901bfbc05c9e3..4085c9b33075e18f7e4e3ad82fdd3fa5f6bb31e2 100644
--- a/lib/node/chunk-CVV3L375.mjs
+++ b/lib/node/chunk-CVV3L375.mjs
@@ -540,6 +540,13 @@ var _NodeClientRequest = class extends ClientRequest {
    * Terminates a pending request.
    */
   terminate() {
+    const tempAgent = this.agent;
+    if (tempAgent && typeof tempAgent === 'object' && typeof tempAgent.destroy !== 'function') {
+      console.log(
+        `No agent to destroy, found '${typeof tempAgent}'->'${typeof tempAgent.destroy}':`,
+        tempAgent
+      );
+    }
     var _a;
     (_a = this.agent) == null ? void 0 : _a.destroy();
   }
diff --git a/lib/node/chunk-SXGRMPXP.js b/lib/node/chunk-SXGRMPXP.js
index a5129f18a9b754acf2bfb3b1545df2378b01a95a..8228b6d4f649937559f45aefbdb8c8df275c5285 100644
--- a/lib/node/chunk-SXGRMPXP.js
+++ b/lib/node/chunk-SXGRMPXP.js
@@ -540,6 +540,13 @@ var _NodeClientRequest = class extends _http.ClientRequest {
    * Terminates a pending request.
    */
   terminate() {
+    const tempAgent = this.agent;
+    if (tempAgent && typeof tempAgent === 'object' && typeof tempAgent.destroy !== 'function') {
+      console.log(
+        `No agent to destroy, found '${typeof tempAgent}'->'${typeof tempAgent.destroy}':`,
+        tempAgent
+      );
+    }
     var _a;
     (_a = this.agent) == null ? void 0 : _a.destroy();
   }
