diff --git a/dist/preview.d.ts b/dist/preview.d.ts
index 73525b5fe07240d27733b59362cf2076cd28a6ed..0420e6b170a65b6d0db64fcfcfa07b68cd851942 100644
--- a/dist/preview.d.ts
+++ b/dist/preview.d.ts
@@ -174,7 +174,7 @@ type UnionToIntersection<Union> = (
 declare function __definePreview<Addons extends PreviewAddon<never>[]>(input: {
     addons: Addons;
 } & ProjectAnnotations<ReactTypes & InferTypes<Addons>>): ReactPreview<ReactTypes & InferTypes<Addons>>;
-interface ReactPreview<T extends AddonTypes> extends Preview<ReactTypes & T> {
+interface ReactPreview<T extends AddonTypes = any> extends Omit<Preview<ReactTypes & T>, 'meta'> {
     meta<TArgs extends Args, Decorators extends DecoratorFunction<ReactTypes & T, any>, TMetaArgs extends Partial<TArgs>>(meta: {
         render?: ArgsStoryFn<ReactTypes & T, TArgs>;
         component?: ComponentType<TArgs>;
@@ -187,16 +187,13 @@ interface ReactPreview<T extends AddonTypes> extends Preview<ReactTypes & T> {
     }>;
 }
 type DecoratorsArgs<TRenderer extends Renderer, Decorators> = UnionToIntersection<Decorators extends DecoratorFunction<TRenderer, infer TArgs> ? TArgs : unknown>;
-interface ReactMeta<T extends ReactTypes, MetaInput extends ComponentAnnotations<T>> extends Meta<T, MetaInput> {
+interface ReactMeta<T extends ReactTypes = any, MetaInput = any> extends Omit<Meta<T, ComponentAnnotations<T, any>>, 'story'> {
     story<TInput extends (() => ReactTypes['storyResult']) | (StoryAnnotations<T, T['args']> & {
         render: () => ReactTypes['storyResult'];
-    })>(story?: TInput): ReactStory<T, TInput extends () => ReactTypes['storyResult'] ? {
-        render: TInput;
-    } : TInput>;
-    story<TInput extends Simplify<StoryAnnotations<T, AddMocks<T['args'], MetaInput['args']>, SetOptional<T['args'], keyof T['args'] & keyof MetaInput['args']>>>>(story?: TInput): ReactStory<T, TInput>;
+    })>(story?: TInput): ReactStory<T>;
+    story<TInput = any>(story?: TInput): ReactStory<T>;
 }
-interface ReactStory<T extends ReactTypes, TInput extends StoryAnnotations<T, T['args']>> extends Story<T, TInput> {
+interface ReactStory<T extends ReactTypes> {
     Component: ComponentType<Partial<T['args']>>;
 }
-
 export { ReactPreview, ReactStory, __definePreview };
\ No newline at end of file
