diff --git a/dist/cjs/CatchBoundary.d.cts b/dist/cjs/CatchBoundary.d.cts
index 6d659e86f69afd0236fad94aad83a12ae685f368..1e7c66be115740093dc058c6c8d21e0b8afdd15a 100644
--- a/dist/cjs/CatchBoundary.d.cts
+++ b/dist/cjs/CatchBoundary.d.cts
@@ -28,7 +28,7 @@ export declare class CatchBoundary extends React.Component<{
     };
     reset: () => void;
     componentDidCatch(error: unknown, errorInfo: ErrorInfo): void;
-    render(): string | number | bigint | boolean | React.ReactElement<unknown, string | React.JSXElementConstructor<any>> | Iterable<React.ReactNode> | Promise<string | number | bigint | boolean | React.ReactPortal | React.ReactElement<unknown, string | React.JSXElementConstructor<any>> | Iterable<React.ReactNode> | null | undefined> | React.FunctionComponentElement<import('@tanstack/router-core').ErrorComponentProps> | null | undefined;
+    render(): React.ReactNode;
 }
 export declare function ErrorComponent({ error }: {
     error: unknown;
diff --git a/dist/esm/CatchBoundary.d.ts b/dist/esm/CatchBoundary.d.ts
index 3f74e54fc00122f3417d351a2119569f98b5905a..086f6b8140268f51c59792a800d215ae0c40b8af 100644
--- a/dist/esm/CatchBoundary.d.ts
+++ b/dist/esm/CatchBoundary.d.ts
@@ -28,7 +28,7 @@ export declare class CatchBoundary extends React.Component<{
     };
     reset: () => void;
     componentDidCatch(error: unknown, errorInfo: ErrorInfo): void;
-    render(): string | number | bigint | boolean | React.ReactElement<unknown, string | React.JSXElementConstructor<any>> | Iterable<React.ReactNode> | Promise<string | number | bigint | boolean | React.ReactPortal | React.ReactElement<unknown, string | React.JSXElementConstructor<any>> | Iterable<React.ReactNode> | null | undefined> | React.FunctionComponentElement<import('@tanstack/router-core').ErrorComponentProps> | null | undefined;
+    render(): React.ReactNode;
 }
 export declare function ErrorComponent({ error }: {
     error: unknown;
