diff --git a/lib/main.d.ts b/lib/main.d.ts
index <oldhash>..<newhash> 100644
--- a/lib/main.d.ts
+++ b/lib/main.d.ts
@@ -1,7 +1,7 @@
-import { ASTNode, Type, AnyType, Field } from "./types";
-import { NodePath } from "./node-path";
+import { ASTNode, type Type, AnyType, Field } from "./types";
+import { type NodePath } from "./node-path";
 import { namedTypes } from "./gen/namedTypes";
-import { builders } from "./gen/builders";
+import { type builders } from "./gen/builders";
 import { Visitor } from "./gen/visitor";