diff --git a/build/watchers/FSEventsWatcher.js b/build/watchers/FSEventsWatcher.js
index 2273a6cbd2110ddbbd1525820289c3d93ab46cac..d637a644239ae4a7db407ce779f43b4dec9a44ed 100644
--- a/build/watchers/FSEventsWatcher.js
+++ b/build/watchers/FSEventsWatcher.js
@@ -103,7 +103,7 @@ function _interopRequireWildcard(obj, nodeInterop) {
 // @ts-ignore: this is for CI which runs linux and might not have this
 let fsevents = null;
 try {
-  fsevents = require('fsevents');
+  // fsevents = require('fsevents');
 } catch {
   // Optional dependency, only supported on Darwin.
 }
